src/zjs_ocf_server.c

```diff
@@ -150,9 +150,10 @@ struct server_resource *new_server_resource(char *path)
     struct server_resource *resource = zjs_malloc(sizeof(struct server_resource));
     memset(resource, 0, sizeof(struct server_resource));
 
-    resource->resource_path = zjs_malloc(strlen(path));
-    memcpy(resource->resource_path, path, strlen(path));
-    resource->resource_path[strlen(path)] = '\0';
+    uint32_t len = strlen(path);
+    resource->resource_path = zjs_malloc(len + 1);
+    strncpy(resource->resource_path, path, len);
+    resource->resource_path[len] = '\0';
```

Comment:
Really you should check results of malloc twice in this function.

And if you do strncpy on len + 1 you apparently wouldn't need to manually set the '\0'. (Verify, but that seems to be what the manpage says.)